magento/magento2
# Set default indexer mode to 'schedule'

### Description
[It is recommended](https://experienceleague.adobe.com/docs/commerce-admin/systems/tools/index-management.html#:~:text=As%20a%20best%20practice%2C%20we%20recommend%20that%20you%20use%20scheduled%20indexing.) that indexers be in the `schedule` ("Update by Schedule") mode, and not `realtime` ("Update on Save") mode. However, when Magento is installed for the first time, all indexers are by default in the "wrong" mode. This pull request changes the default for new indexers to be in the expected / recommended mode.

This change will not impact any existing indexers and therefore existing websites should not notice any impact from this change. If a module with an indexer is installed on an existing website, it will be created with the new default mode: `schedule`.

### Manual testing scenarios
1. Install Magento anew
1. Navigate in the admin to System -> Tools -> Index Management
1. Notice the colour of the "Mode" column. Before this change, these were all red, indicating an error/problem; after this change these are all green, indicating that all is well.
1. Install a module which adds indexers (such as `algolia/algoliasearch-magento-2`)
1. Navigate in the admin to System -> Tools -> Index Management
1. Notice the colour of the "Mode" column. Before this change, new indexers were all red, indicating an error/problem; after this change these are all green, indicating that all is well.

### Contribution checklist


### Resolved issues:
1. [x] resolves magento/magento2#36419: Set default indexer mode to 'schedule'
